server/resp.go

```diff
@@ -125,15 +125,7 @@ func (c *Conn) writeResultset(r *Resultset) error {
 		return err
 	}
 
-	for _, v := range r.Fields {
-		data = data[0:4]
-		data = append(data, v.Dump()...)
-		if err := c.WritePacket(data); err != nil {
-			return err
-		}
-	}
-
-	if err := c.writeEOF(); err != nil {
+	if err := c.writeFieldList(r.Fields); err != nil {
```

Comment:
This variant adds a new memory allocation because do not reuses `data`. 

How about a buffer transfer to `writeFieldList`?

Reply:
good point